dags/inference/maxtext_inference_offline_benchmark.py

```diff
@@ -0,0 +1,199 @@
+    "power_begin", "power_end", "result_validity",
+    "early_stopping_ttft_result", "early_stopping_tpot_result"
+  }
+
+  metrics = {}
+  dimensions = {}
+
+  with open(log_file_path, "r") as f:
+    log_content = f.read()
+
+  log_pattern = r":::MLLOG ({.*})"
+  for line in log_content.split("\n"):
+    match = re.search(log_pattern, line)
+    if match:
+      try:
+        entry = json.loads(match.group(1))
+        key = entry.get("key", "")
+        value = entry.get("value")
+
+        if isinstance(value, (int, float)):
+          metrics[key] = value
+        elif key in dimension_keys:
+          dimensions[key] = value
+      except json.JSONDecodeError:
+        continue
+
+  result = {"metrics": metrics, "dimensions": dimensions}
+
+  with jsonlines.open(output_path, mode="w") as writer:
+    writer.write(result)
+
+  return result
+
+if __name__ == "__main__":
+    convert_mlperf_log_to_jsonlines("./mlperf_log_detail.txt", "metric_report.jsonl")
+EOL"""
+
+
+def maxtext_inference_offline_benchmark_config(
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    test_mode: common.SetupMode,
+    project_name: str = PROJECT_NAME,
+    runtime_version: str = RUNTIME_IMAGE,
+    network: str = "default",
+    subnetwork: str = "default",
+    is_tpu_reserved: bool = True,
+    num_slices: int = 1,
+    maxtext_branch: str = "",
+):
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.BENCHMARK_DATASET,
+  )
+  git_clone_maxtext = "git clone https://github.com/google/maxtext.git"
+  if maxtext_branch:
+    git_clone_maxtext += f" -b {maxtext_branch}"
+
+  set_up_cmds = (
+      "pip install --upgrade pip",
+      "sudo apt-get -y update",
+      "sudo apt-get -y install python3.10-venv",
+      "sudo apt-get -y install jq",
+      "python -m venv .env",
+      "source .env/bin/activate",
+      # Setup MaxText
+      git_clone_maxtext,
+      f"cd maxtext && bash setup.sh MODE={test_mode.value} && cd ..",
+      "pip install torch --index-url https://download.pytorch.org/whl/cpu",
+      # Setup Loadgen
+      "git clone https://github.com/mlcommons/inference.git",
+      "cd inference/loadgen && pip install . && cd ../..",
+  )
+
+  run_model_cmds = (
+      "source .env/bin/activate",
+      "cd maxtext/MaxText/inference_mlperf/trillium",
+      "gsutil cp gs://cloud-tpu-inference-public/mlcommons/inference/language/llama2-70b/data/processed-openorca/open_orca_gpt4_tokenized_llama.sampled_24576.pkl /tmp/processed-data.pkl",
+      "export DATA_DISK_DIR=/tmp",
+      "export CHECKPOINT=gs://inference-benchmarks/models/llama2-70b-chat/quant/int8_",
+      "export TOKENIZER_PATH=/home/ml-auto-solutions/maxtext/assets/tokenizer.llama2",
+      "echo $TOKENIZER_PATH",
+      "bash benchmarks_llama2-70b-trillium_2x4.sh -x -t -s",
```

Comment:
We should use full benchmarks here - not the -t option

Reply:
We should also uncomment accuracy and audit run in above script.

Reply:
I have changed the MaxText script to run all tests by default, but also supports selecting a single test.

Reply:
Great thanks.